kanidmd/src/lib/actors/v1_read.rs

```diff
@@ -167,6 +172,38 @@ impl QueryServerReadV1 {
         res
     }
 
+    pub async fn handle_online_backup(&self, msg: LiveBackupEvent, outpath: &str) {
+        let mut audit = AuditScope::new("online backup", msg.eventid, self.log_level);
+
+        ltrace!(audit, "Begin online backup event {:?}", msg.eventid);
+
+        // TODO: might be something like "backup-timestamp.json"
+        let now: DateTime<Utc> = Utc::now();
+        let timestamp = now.to_rfc3339_opts(SecondsFormat::Secs, true);
+        let dest_file = format!("{}/backup-{}.json", outpath, timestamp);
```

Comment:
Need to ensure that outpath exists and is a directory before this too. Should also check that dest_file does NOT exist already.

Reply:
Yeah, checking the path is good. The function that writes the backup will overwrite existing files. I think this is okay, especially with timestamp, or?

Reply:
(wanted to post that here) Still todo, should we create the path if it is not there? I think we should try and fail if we cannot.

Reply:
I think overwriting should fail, because every time stamp should be new, so we should never stomp something that exists. 

i think we should create the outhpath folder if it doesn't exist (and set it's permissions properly) and fail if doesn't exist or can't be created.

Reply:
This should be done now.